SYCL/BFloat16/bfloat16_builtins.cpp

```diff
@@ -1,11 +1,13 @@
-// REQUIRES: cuda
+// RUN: %if cuda %{%clangxx -fsycl -fsycl-targets=%sycl_triple -Xsycl-target-backend --cuda-gpu-arch=sm_80 %s -o %t.out %}
+// RUN: %if cuda %{%GPU_RUN_PLACEHOLDER %t.out %}
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// TODO currently the feature isn't supported on FPGA.
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUNx: %ACC_RUN_PLACEHOLDER %t.out
 //
-// Currently this test fails to compile for backends other than cuda.
-// Other backends could use this test when bfloat16 math function support is
-// added.
-//
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out -Xsycl-target-backend --cuda-gpu-arch=sm_80
-// RUN: %t.out
+// Not currently supported on HIP.
```

Comment:
HIP should be supported by the generic impls. You've commented out the UNSUPPORTED: hip anyway, I guess you meant to remove this comment? Otherwise LGTM

Reply:
Hi, @JackAKirk 
The generic imp should work on HIP backend but I don't have HIP environment for testing, so disable it on HIP temporarily.
Thanks very much.

Reply:
> Hi, @JackAKirk The generic imp should work on HIP backend but I don't have HIP environment for testing, so disable it on HIP temporarily. Thanks very much.

But if you remove the unsupported you can test it on HIP with the CI by linking with the intel/llvm PR using this command https://github.com/intel/llvm/pull/7959#issuecomment-1375918932, no? Then you don't really need to test it locally. If it is generic I don't know why it would not work on HIP.